tests/utils/spark_operator.go

```diff
@@ -115,6 +136,49 @@ func (spark *SparkOperatorInstallation) getInstanceStatus() (string, error) {
 	return status, err
 }
 
+func (spark *SparkOperatorInstallation) WaitForJobState(job SparkJob, state v1beta2.ApplicationStateType, duration time.Duration) error {
+	log.Infof("Waiting for spark application %s to reach %s state", job.Name, state)
+	err := Retry(duration, 1*time.Second, func() error {
+		app, err := spark.SparkClients.SparkoperatorV1beta2().SparkApplications(spark.Namespace).Get(job.Name, v1.GetOptions{})
+		if err != nil {
+			return err
+		} else if app.Status.AppState.State != state {
+			return errors.New(fmt.Sprintf("%s state is %s", job.Name, app.Status.AppState.State))
+		}
+		return nil
+	})
+
+	if err == nil {
+		log.Infof("Spark application %s is now %s", job.Name, state)
+	}
+
+	return err
+}
+
+func (spark *SparkOperatorInstallation) JobExecutors(job SparkJob) (map[string]v1beta2.ExecutorState, error) {
```

Comment:
function name looks confusing given that the returned entity is `ExecutorState`. Is the function a getter? Let's make this fact explicit if so

Reply:
renamed the method